internal/handler/handlers.go

```diff
@@ -185,6 +185,97 @@ func SaveRecruiterProfileHandler(svr server.Server, recRepo *recruiter.Repositor
 	}
 }
 
+func SaveDeveloperMetadataHandler(svr server.Server, devRepo *developer.Repository) http.HandlerFunc {
+	return middleware.UserAuthenticatedMiddleware(
+		svr.SessionStore,
+		svr.GetJWTSigningKey(),
+		func(w http.ResponseWriter, r *http.Request) {
+			req := &struct {
+				DeveloperProfileID string  `json:"developer_profile_id"`
+				MetadataType       string  `json:"metadata_type"`
+				Title              string  `json:"title"`
+				Description        string  `json:"description"`
+				Link               *string `json:"link,omitempty"`
+			}{}
+
+			if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+				svr.Log(errors.New("invalid developer metadata"), "invalid developer metadata")
+				svr.JSON(w, http.StatusBadRequest, nil)
+				return
+			}
+			_, err := middleware.GetUserFromJWT(r, svr.SessionStore, svr.GetJWTSigningKey())
```

Comment:
we need to make sure that whoever is updating the metadata is either
- user updating his own metadata
- admin can update anyones metadata

You can use the UserJWT coming from this function and get the user id and user type

so something like this

if user.Type != admin && user.ID != req.DeveloperPRofileID { return 401 }

Reply:
Done.